pkgs/build-support/fetchgit/default.nix

```diff
@@ -94,6 +99,9 @@ stdenvNoCC.mkDerivation {
     export HOME=$PWD
   '';
 
+  # if verifyTag enabled assume rev to be a tag and include it in the .git directory
+  NIX_PREFETCH_GIT_CHECKOUT_HOOK = if verifyTag then ''clean_git -C "$dir" fetch origin "refs/tags/$rev:refs/tags/$rev"'' else null;
```

Comment:
This feels pretty hacky so I'd be glad for any ideas of how to improve this.
In my view it should be the default to include the passed reference/tag in the .git output, but changing this now would break backwards-reproducibility.
Maybe this could be an additionall `--fetch-tag` flag for nix-prefetch-git?